### ISSUE TITLE
current main branch fails make -j distcheck

### ISSUE THREAD

**Issue by edwardhartnett** [Unidata/netcdf-c#2544]
When I build like:

```
ed@koko:~/netcdf-c$ export LD_LIBRARY_PATH=/usr/local/hdf5-1.12.2/lib
ed@koko:~/netcdf-c$ export CPPFLAGS="-I/usr/local/hdf5-1.12.2/include -I/usr/local/szip-2.1.1/include"
ed@koko:~/netcdf-c$ export LDFLAGS="-L/usr/local/hdf5-1.12.2/lib -L/usr/local/szip-2.1.1/lib"
ed@koko:~/netcdf-c$ export DISTCHECK_CONFIGURE_FLAGS=--disable-dap-remote-tests
ed@koko:~/netcdf-c$ ./configure --disable-dap-remote-tests

```
Then make -j check works. (Yea!)

But make -j distcheck does not work:

```
make[5]: Entering directory '/home/ed/netcdf-c/netcdf-c-4.9.1-development/_build/sub/dap4_test'
FAIL: test_meta.sh
FAIL: test_parse.sh
FAIL: test_raw.sh
FAIL: test_fillmismatch.sh
FAIL: test_data.sh

```
Note that I set DISTCHECK_CONFIGURE_FLAGS to --disable-dap-remote-tests, so should these tests even be running?

**Comment by edwardhartnett** [comment]
I also fails make distcheck, so this is not a parallel build/test issue.

When I look in the log for the failed tests I see:

```
../../../dap4_test/test_parse.sh: 27: cd: can't cd to /home/ed/netcdf-c/netcdf-c-4.9.1-development/_build/sub/dap4_test/baselinethredds
FAIL test_parse.sh (exit status: 2)

```

The problem here is the dap4_test/baseline (and perhaps other directories under dap4_test) does not have a Makefile.am.

If you just want to distribute the files in this directory for testing, we need a Makefile.am which lists all the files as EXTRA_DIST.

I will submit a PR and see if that fixes my distcheck...

**Comment by edwardhartnett** [comment]
...BTW it seems like this problem would also be present in the tarball release. That is, the tarball release, if constructed with make dist, will only contain the files listed in the Makefile.am files.

In fact, I see that make dist also breaks:

```
make[4]: Entering directory '/home/ed/netcdf-c/dap4_test'
 (cd baseline && make  top_distdir=../../netcdf-c-4.9.1-development distdir=../../netcdf-c-4.9.1-development/dap4_test/baseline \
     am__remove_distdir=: am__skip_length_check=: am__skip_mode_fix=: distdir)
make[5]: Entering directory '/home/ed/netcdf-c/dap4_test/baseline'
make[5]: *** No rule to make target 'distdir'.  Stop.
make[5]: Leaving directory '/home/ed/netcdf-c/dap4_test/baseline'
make[4]: *** [Makefile:1129: distdir-am] Error 1
make[4]: Leaving directory '/home/ed/netcdf-c/dap4_test'
make[3]: *** [Makefile:1125: distdir] Error 2
make[3]: Leaving directory '/home/ed/netcdf-c/dap4_test'
make[2]: *** [Makefile:873: distdir-am] Error 1
make[2]: Leaving directory '/home/ed/netcdf-c'
make[1]: *** [Makefile:867: distdir] Error 2
make[1]: Leaving directory '/home/ed/netcdf-c'
make: *** [Makefile:968: dist] Error 2
ed@koko:~/netcdf-c$ 

```
@DennisHeimbigner unfortunately we cannot list subdirectories in EXTRA_DIST, only files. ;-)

**Comment by edwardhartnett** [comment]
I have a fix to this but I have to turn off nczarr_test/run_scalar.sh for it to work. See #2545

**Comment by DennisHeimbigner** [comment]
I am doing a distcheck right now, and it looks like I can
put directories in EXTRA_DIST.
There was, however, an error in d4test_common.sh that specified
an incorrect path to one of those directories.

**Comment by WardF** [comment]
This path issue is fixed in the 4.9.1 wellspring branch and will be back propagated to main in due time. I'll take a look at doing that early, but no changes to fix it are needed now. I'll be back in town next Tuesday and will bring this fix to Main if nothing else.

**Comment by DWesl** [comment]
Putting globs or directories in an Autotools file list is discouraged, but putting files in a subdirectory is perfectly allowed, and sometimes encouraged.

**Comment by DennisHeimbigner** [comment]
I am surprised; Autotools seems to handle directories with no problems.
What is the argument against them?

**Comment by DWesl** [comment]
For wildcards/globs, there's [this section of the documentation explaining the reasons](https://www.gnu.org/software/automake/manual/html_node/Wildcards.html); requesting tests in PRs and running tests in CI seem to cover most of their concerns.  The one exception is lack of portability to non-GNU make, which may not be a concern here.  

I wasn't aware that Autotools accepted directories in `EXTRA_DIST`, but that gets described [just before this bit of the documentation](https://www.gnu.org/software/automake/manual/html_node/Basics-of-Distribution.html#index-EXTRA_005fDIST).  The Autotools developers also recommend against using directories in `EXTRA_DIST` due to the likelihood of picking up extraneous files, unless you add something to `dist-hook` to take the most likely extraneous files back out. If the release archive is created on CI rather than an active developer's computer, there are fewer places for extra files to creep into the directory.

In short: it looks like I failed to consider the Autotools advice in the context of CI.
